### ISSUE TITLE
Add a reference implementation for Wide ResNet

### ISSUE THREAD

**Issue by nalzok** [google/flax#2400]
Currently, there are no example implementations for Wide ResNet in Flax:

+ We have [n2cholas/jax-resnet](https://github.com/n2cholas/jax-resnet), but it simply scales a ResNet up in width instead of following the parameterization of [the original paper](https://arxiv.org/abs/1605.07146). For example, it's not obvious how to make a ResNet-28-10 with this repository.
+ We have [google-research/google-research](https://github.com/google-research/google-research), but the example has been [removed](https://github.com/google-research/google-research/commit/48d62a0d06dbd58fe1407f9106817455365156c7) because [it was using the old API and we had nobody actively working on it and willing to port it](https://github.com/google/flax/issues/122#issuecomment-1091397831). I tried to [upgrade](https://flax.readthedocs.io/en/latest/advanced_topics/linen_upgrade_guide.html) it, but without success since I am not familiar with the old API.
+ We have [deepmind/deepmind-research](https://github.com/deepmind/deepmind-research), which does include an implementation of Wide ResNet in [`model_zoo.py`](https://github.com/deepmind/deepmind-research/blob/master/adversarial_robustness/jax/model_zoo.py), but it's in Haiku instead of Flax.

I think it's worth having a reference implementation for Wide ResNet in `examples/`, since it's a quite common baseline model.

**Comment by cgarciae** [comment]
We are currently avoiding adding too many examples to `examples/` as it increases the maintenance burden. Wondering if this could added to [n2cholas/jax-resnet](https://github.com/n2cholas/jax-resnet) as we already reference it in our [Community Examples](https://github.com/google/flax/tree/main/examples#community-examples).

**Comment by marcvanzee** [comment]
I agree with @cgarciae, we already have a Resnet, and having more specific Resnet is something that fits better in dedicated libraries. We are always happy to link to examples of course!
